### ISSUE TITLE
Always load first channel from multichannel audio/WTs

### ISSUE THREAD

**Issue by mkruselj** [surge-synthesizer/surge#5726]
Instead of throwing an error requiring a mono audio file. Obviously, patch up the error message while at it etc.

**Comment by baconpaul** [comment]
All the action here is in `src/common/WavFileSupport.cpp`. The steps are

1. Around line 157, no longer error out if numChannels != 1. Fine.
2. Similar the error condition at 448 needs to not be channel dependent

So then you will get to line 462 with a stereo data block. At this point all hell breaks loose. Well, not quite, but almost. 

So what you need to do is replace this 

```
 if (wavdata && wt)
    {
        waveTableDataMutex.lock();
        wt->BuildWT(wavdata, wh, wh.flags & wtf_is_sample);
        waveTableDataMutex.unlock();
        free(wavdata);
    }
```

with

```
 if (wavdata && wt)
    {
        if (numChannels == 1)
        {
            waveTableDataMutex.lock();
            wt->BuildWT(wavdata, wh, wh.flags & wtf_is_sample);
            waveTableDataMutex.unlock();
        }
        else
        {
               char *leftChannel;
               // tricky stuff here below

               waveTableDataMutex.lock();
               wt->BuildWT(leftChannel, wh, wh.flags & wtf_is_sample);
               waveTableDataMutex.unlock();

               free(leftChannel);
        }
        free(wavdata);
    }
```

OK so what is that tricky bit? Well if i recall wav files are interleaved sample c1 c2 sample c1 c2. So my guess is that tricky bit looks something like this

```
memsize = datasamples * bitsPerSample / 8;
leftChannel = (char *)malloc(memsize * sizeof(char));
 for (size_t i=0; i<datasamples; ++i)
 {
     size_t destAddr = i * bitsPerSample;
     size_t srcAddr = i * bitsPerSample * numChannels;
     std::memcpy(leftChannel + destAddr, wavdata+srcAddr, bitsPerSample);
 }
```

basically copying out the strided src data into the channel.

**Comment by mkruselj** [comment]
Awesome!

And if I understand the tricky bit correctly, this also covers the multichannel scenario, not just stereo. 👍

**Comment by baconpaul** [comment]
Yup

**Comment by baconpaul** [comment]
Oh and the memcpy should be bits per sample / 8
